xrdp/xrdp_mm.c

```diff
@@ -2464,6 +2467,39 @@ xrdp_mm_trans_process_drdynvc_data(struct xrdp_mm *self,
     return 0;
 }
 
+/*****************************************************************************/
+/* Acknowledgement from chansrv that Unicode input is supported
+ */
+static int
+xrdp_mm_trans_process_unicode_ack(struct xrdp_mm *self,
+                                  struct stream *s)
+{
+    int status;
+    if (!s_check_rem(s, 4))
+    {
+        return 1;
+    }
+    in_uint32_le(s, status);
+    switch (status)
+    {
+        case 0:
+            LOG(LOG_LEVEL_INFO, "Chansrv is handling Unicode input");
+            self->wm->client_info->unicode_input_support = UIS_ACTIVE;
+            break;
+
+        case 1:
+            LOG(LOG_LEVEL_INFO, "Chansrv does not support Unicode input");
+            break;
+
+        default:
```

Comment:
Better specify 2 as init failure as it does in line "chansrc:893" here. Using default is unclear.

Reply:
Agreed.

Reply:
I can do this if you are short on time.

Reply:
I'm on it right now...